### PR TITLE
Extract locale data

### LINKED ISSUE #418: Setup locale extraction
Should extract strings from the core app and the app specified by appName in the config.

### PR DESCRIPTION
**Pull request by muffinresearch**
Fixes #418 

* Uses sprintf-js for placeholder subs 
* I forked babel-gettext-extractor and patched it for the following:
  * 🐃  Fix for undefined translation keys which is what caused the empty output see https://github.com/muffinresearch/babel-gettext-extractor/commit/54991908110858db6f545dec9907fe2672293904
  * 🐃 Fix for lack of content output idempotency. This is done by key sorting (which is a bit of a hack) but looks to work in practice. See https://github.com/muffinresearch/babel-gettext-extractor/commit/f0f00b2afb71cba5edfb43d377bde9e1b08cdb46
* Made the creation date dynamic
* Used a babel plugin for dedenting template literals so it gets pre-processed and is therefore correct in the translation.

Next up building some locales with the templates and using them...

Obviously we'll need to clean up and decide how to proceed wrt the plugin fork.

### CODE REVIEW

File: locale/templates/LC_MESSAGES/search.pot [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
+"Language-Team: LANGUAGE <LL@li.org>\n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=utf-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+"Plural-Forms: nplurals=2; plural=(n!=1);\n"
+"Content-Type: text/plain; charset=utf-8\n"
+
+#: src/core/components/LoginPage/index.js:13
+msgid "Login Required"
```
Comment (mstriemer) [inline locale/templates/LC_MESSAGES/search.pot]: That's awesome that this string isn't in the disco one 👍

File: webpack.l10n.config.babel.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+const babelrc = fs.readFileSync('./.babelrc');
+const babelrcObject = JSON.parse(babelrc);
+const babelPlugins = babelrcObject.plugins || [];
+
+// Create UTC creation date in the correct format.
+const potCreationDate = new Date().toISOString()
+  .replace('T', ' ')
+  .replace(/:\d{2}.\d{3}Z/, '+0000');
```
Comment (mstriemer) [inline webpack.l10n.config.babel.js]: No seconds?

File: webpack.l10n.config.babel.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      'PO-Revision-Date': 'YEAR-MO-DA HO:MI+ZONE',
+      'Last-Translator': 'FULL NAME <EMAIL@ADDRESS>',
+      'Language-Team': 'LANGUAGE <LL@li.org>',
+      'MIME-Version': '1.0',
+      'Content-Type': 'text/plain; charset=utf-8',
+      'Content-Transfer-Encoding': '8bit',
+      'plural-forms': 'nplurals=2; plural=(n!=1);',
+    },
+    functionNames: {
+      _: ['msgid'],
```
Comment (mstriemer) [inline webpack.l10n.config.babel.js]: There's at least one ngettext call in there already, I'm guessing that gets setup automatically? It's using a raw replace which should be updated to use `sprintf` I think. In src/search/components/SearchResult/index.js.

File: webpack.l10n.config.babel.js (hunk 2 above)
Comment (muffinresearch) [inline webpack.l10n.config.babel.js]: This is the format expected in the pot file, I borrowed this from i18n-abide. Looks to match-up to the format we are using the the django pot files and it should be in UTC hence the +0000 is correct.

File: webpack.l10n.config.babel.js (hunk 3 above)
Comment (mstriemer) [inline webpack.l10n.config.babel.js]: Actually I don't see the string so it seems to be missing.

**Comment by mstriemer** [comment]
r+wc here. I'm going to look over the patches to the extractor too but we could merge those later if I have any comments.

File: webpack.l10n.config.babel.js (hunk 3 above)
Comment (muffinresearch) [inline webpack.l10n.config.babel.js]: Ah I might have missed that one. We could use some warnings in the plugin for that. I'll look for it.

Comment (muffinresearch) [inline webpack.l10n.config.babel.js]: OK I'll see about adding that to the conf and it should find it.

**Pull request merged**